File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.validator.js
let JkValidatorDataValidator = function() {};

JkValidatorDataValidator.prototype._t = { "JkValidatorDataValidator" : true };
JkValidatorDataValidator.prototype._tobj = JkValidatorDataValidator;

JkValidatorDataValidator.NEW = function() {
	var v = new JkValidatorDataValidator;
	return v.CTOR_JkValidatorDataValidator();
};

JkValidatorDataValidator.prototype.CTOR_JkValidatorDataValidator = function() {
	return this;
};

JkValidatorDataValidator.isEmailAddressValid = function(emailAddress) {
	if(JkLangString.isEmpty(emailAddress)) {
		return false;
	}
	if(JkLangString.getLength(emailAddress) > 254) {
		return false;
	}
	var ci = JkLangString.iterate(emailAddress);
	if(!(ci != null)) {
		return false;
	}
	var insideQuotation = false;
	var outsideQuotation = false;
	var dotPreceded = false;
	var isFirstCharacter = true;
	var containsAtCharacter = false;
	var i = -1;
	while(true){
		i++;
		var c = ci.getNextChar();
		if(c < 1) {
			break;
		}
		if(outsideQuotation) {
			if(c != 46 && c != 64) {
				return false;
			}
		}
		if(c >= 65 && c <= 90) {
			;
		}
		else if(c >= 97 && c <= 122) {
			;
		}
		else if(c >= 48 && c <= 57) {
			;
		}
		else if(c == 33) {
			;
		}
		else if(c >= 35 && c <= 39) {
			;
		}
		else if(c == 42) {
			;
		}
		else if(c == 43) {
			;
		}
		else if(c == 45) {
			;
		}
		else if(c == 47) {
			;
		}
		else if(c == 61) {
			;
		}
		else if(c == 63) {
			;
		}
		else if(c >= 94 && c <= 96) {
			;
		}
		else if(c >= 123 && c <= 126) {
			;
		}
		else if(c == 46) {
			if(isFirstCharacter) {
				return false;
			}
			if(dotPreceded && !insideQuotation) {
				return false;
			}
			dotPreceded = true;
			continue;
		}
		else if(c == 32) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 34) {
			if(insideQuotation) {
				insideQuotation = false;
				outsideQuotation = true;
			}
			else {
				if(!isFirstCharacter && !dotPreceded) {
					return false;
				}
				insideQuotation = true;
				outsideQuotation = false;
			}
		}
		else if(c == 40) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 41) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 44) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 58) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 59) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 60) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 62) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 64) {
			if(isFirstCharacter) {
				return false;
			}
			if(!insideQuotation) {
				if(dotPreceded) {
					return false;
				}
				containsAtCharacter = true;
				break;
			}
		}
		else if(c == 91) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 92) {
			if(!insideQuotation) {
				return false;
			}
		}
		else if(c == 93) {
			if(!insideQuotation) {
				return false;
			}
		}
		else {
			return false;
		}
		isFirstCharacter = false;
		dotPreceded = false;
	}
	if(i < 0) {
		return false;
	}
	if(i > 64) {
		return false;
	}
	if(!containsAtCharacter) {
		return false;
	}
	var domain = JkLangString.getEndOfString(emailAddress, (i + 1));
	if(JkLangString.isEmpty(domain)) {
		return false;
	}
	if(JkLangString.getLength(domain) > 253) {
		return false;
	}
	if(JkLangString.startsWith(domain, "[", 0)) {
		if(!JkLangString.endsWith(domain, "]")) {
			return false;
		}
		var ip = JkLangString.getSubString(domain, 1, (JkLangString.getLength(domain) - 2));
		if(JkLangString.isEmpty(ip)) {
			return false;
		}
		var nodes = JkLangString.split(ip, '.'.charCodeAt(), 0);
		if(!(nodes != null) || JkLangVector.getSize(nodes) != 4) {
			return false;
		}
		if(nodes != null) {
			var n2 = 0;
			var m = nodes.length;
			for(n2 = 0 ; n2 < m ; n2++) {
				var node = nodes[n2];
				if(node != null) {
					ci = JkLangString.iterate(node);
					while(true){
						var c1 = ci.getNextChar();
						if(c1 < 1) {
							break;
						}
						if(c1 >= 48 && c1 <= 57) {
							;
						}
						else {
							return false;
						}
					}
					var n = JkLangString.toInteger(node);
					if(n < 0 || n > 255) {
						return false;
					}
				}
			}
		}
		return false;
	}
	var domains = JkLangString.split(domain, '.'.charCodeAt(), 0);
	if(JkLangVector.getSize(domains) <= 1) {
		return false;
	}
	if(domains != null) {
		var n3 = 0;
		var m2 = domains.length;
		for(n3 = 0 ; n3 < m2 ; n3++) {
			var label = domains[n3];
			if(label != null) {
				if(JkLangString.isEmpty(label)) {
					return false;
				}
				if(JkLangString.getLength(label) > 63) {
					return false;
				}
				ci = JkLangString.iterate(label);
				if(!(ci != null)) {
					return false;
				}
				var hyphenPreceded = false;
				isFirstCharacter = true;
				while(true){
					var c2 = ci.getNextChar();
					if(c2 < 1) {
						break;
					}
					if(c2 >= 65 && c2 <= 90) {
						;
					}
					else if(c2 >= 97 && c2 <= 122) {
						;
					}
					else if(c2 >= 48 && c2 <= 57) {
						;
					}
					else if(c2 == 45) {
						if(isFirstCharacter) {
							return false;
						}
						hyphenPreceded = true;
						isFirstCharacter = false;
						continue;
					}
					else {
						return false;
					}
					hyphenPreceded = false;
					isFirstCharacter = false;
				}
				if(hyphenPreceded) {
					return false;
				}
			}
		}
	}
	return true;
};

JkValidatorDataValidator.checkPassword = function(password, minimum, maximun) {
	if(JkLangString.isEmpty(password)) {
		return JkLangError.forCode("noPassword", null);
	}
	var min = minimum;
	if(min < 8) {
		min = 8;
	}
	if(JkLangString.getLength(password) < min) {
		return JkLangError.forCode("passwordLengthLessThanMinimum", null);
	}
	var max = maximun;
	if(max < 64) {
		max = 64;
	}
	if(JkLangString.getLength(password) > max) {
		return JkLangError.forCode("passwordLengthMoreThanMaximum", null);
	}
	var hasLowerAlpha = false;
	var hasUpperAlpha = false;
	var hasNumeric = false;
	var hasSpecial = false;
	var previousChar = '\0'.charCodeAt();
	var sChar = '\0'.charCodeAt();
	var count = 1;
	var ci = JkLangString.iterate(password);
	while(true){
		var c = ci.getNextChar();
		if(c < 1) {
			break;
		}
		if(c > 32 && c < 127) {
			if(previousChar == c) {
				count++;
				if(count == 3) {
					return JkLangError.forCode("passwordHasRepetitiveCharacters", null);
				}
				previousChar = c;
			}
			else if(c == previousChar + 1) {
				sChar = c;
			}
			else if(c == sChar + 1) {
				return JkLangError.forCode("passwordHasSequentialCharacters", null);
			}
			else {
				previousChar = c;
			}
			if(c >= 65 && c <= 90) {
				hasUpperAlpha = true;
			}
			else if(c >= 97 && c <= 122) {
				hasLowerAlpha = true;
			}
			else if(c >= 48 && c <= 57) {
				hasNumeric = true;
			}
			else {
				hasSpecial = true;
			}
			continue;
		}
		return JkLangError.forCode("passwordHasUnknownCharacter", null);
	}
	if(hasUpperAlpha && hasLowerAlpha == false && hasNumeric == false && hasSpecial == false) {
		return JkLangError.forCode("passwordHasNoAlphanumericCharacterCombination", null);
	}
	else if(hasLowerAlpha && hasUpperAlpha && hasNumeric == false && hasSpecial == false) {
		return JkLangError.forCode("passwordHasNoNumericAndSpecialCharacterCombination", null);
	}
	else if(hasNumeric == false) {
		return JkLangError.forCode("passwordHasNoNumericCharacter", null);
	}
	else if(hasLowerAlpha == false) {
		return JkLangError.forCode("passwordHasNoLowercase", null);
	}
	else if(hasUpperAlpha == false) {
		return JkLangError.forCode("passwordHasNoUppercase", null);
	}
	else if(hasSpecial == false) {
		return JkLangError.forCode("passwordHasNoSpecialCharacter", null);
	}
	else {
		return null;
	}
};

JkValidatorDataValidator.isNameValid = function(name) {
	if(JkLangString.isEmpty(name)) {
		return false;
	}
	var ci = JkLangString.iterate(name);
	var isFirstCharacter = true;
	var spacePreceded = false;
	var hyphenPreceded = false;
	while(true){
		var c = ci.getNextChar();
		if(c < 1) {
			break;
		}
		if(c == 32) {
			if(isFirstCharacter) {
				return false;
			}
			if(spacePreceded) {
				return false;
			}
			spacePreceded = true;
			continue;
		}
		if(c == 45) {
			if(isFirstCharacter) {
				return false;
			}
			if(hyphenPreceded) {
				return false;
			}
			hyphenPreceded = true;
			continue;
		}
		isFirstCharacter = false;
		spacePreceded = false;
		hyphenPreceded = false;
		if(c == 164 || c == 165 || c >= 65 && c <= 90 || c >= 97 && c <= 122) {
			continue;
		}
		return false;
	}
	if(spacePreceded) {
		return false;
	}
	if(hyphenPreceded) {
		return false;
	}
	return true;
};

JkValidatorDataValidator.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkValidatorDataValidator"] === true;
};

File: .sling/libs/sushi/eqela/jk-data/0.27.0/jk.validator.lua
jk = jk or {}

jk.validator = jk.validator or {}

jk.validator.DataValidator = {}
jk.validator.DataValidator.__index = jk.validator.DataValidator
_vm:set_metatable(jk.validator.DataValidator, {})

function jk.validator.DataValidator._create()
	local v = _vm:set_metatable({}, jk.validator.DataValidator)
	return v
end

function jk.validator.DataValidator:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.validator.DataValidator'
	self['_isType.jk.validator.DataValidator'] = true
end

function jk.validator.DataValidator:_construct0()
	jk.validator.DataValidator._init(self)
	return self
end

function jk.validator.DataValidator:isEmailAddressValid(emailAddress)
	if _g.jk.lang.String:isEmpty(emailAddress) then
		do return false end
	end
	if _g.jk.lang.String:getLength(emailAddress) > 254 then
		do return false end
	end
	do
		local ci = _g.jk.lang.String:iterate(emailAddress)
		if not (ci ~= nil) then
			do return false end
		end
		do
			local insideQuotation = false
			local outsideQuotation = false
			local dotPreceded = false
			local isFirstCharacter = true
			local containsAtCharacter = false
			local i = -1
			while true do
				do i = i + 1 end
				do
					local c = ci:getNextChar()
					if c < 1 then
						do break end
					end
					if outsideQuotation then
						if c ~= 46 and c ~= 64 then
							do return false end
						end
					end
					if c >= 65 and c <= 90 then
					elseif c >= 97 and c <= 122 then
					elseif c >= 48 and c <= 57 then
					elseif c == 33 then
					elseif c >= 35 and c <= 39 then
					elseif c == 42 then
					elseif c == 43 then
					elseif c == 45 then
					elseif c == 47 then
					elseif c == 61 then
					elseif c == 63 then
					elseif c >= 94 and c <= 96 then
					elseif c >= 123 and c <= 126 then
					elseif c == 46 then
						if isFirstCharacter then
							do return false end
						end
						if dotPreceded and not insideQuotation then
							do return false end
						end
						dotPreceded = true
						goto _continue1
					elseif c == 32 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 34 then
						if insideQuotation then
							insideQuotation = false
							outsideQuotation = true
						else
							if not isFirstCharacter and not dotPreceded then
								do return false end
							end
							insideQuotation = true
							outsideQuotation = false
						end
					elseif c == 40 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 41 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 44 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 58 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 59 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 60 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 62 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 64 then
						if isFirstCharacter then
							do return false end
						end
						if not insideQuotation then
							if dotPreceded then
								do return false end
							end
							containsAtCharacter = true
							do break end
						end
					elseif c == 91 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 92 then
						if not insideQuotation then
							do return false end
						end
					elseif c == 93 then
						if not insideQuotation then
							do return false end
						end
					else
						do return false end
					end
					isFirstCharacter = false
					dotPreceded = false
				end
				::_continue1::
			end
			if i < 0 then
				do return false end
			end
			if i > 64 then
				do return false end
			end
			if not containsAtCharacter then
				do return false end
			end
			do
				local domain = _g.jk.lang.String:getEndOfString(emailAddress, i + 1)
				if _g.jk.lang.String:isEmpty(domain) then
					do return false end
				end
				if _g.jk.lang.String:getLength(domain) > 253 then
					do return false end
				end
				if _g.jk.lang.String:startsWith(domain, "[", 0) then
					if not _g.jk.lang.String:endsWith(domain, "]") then
						do return false end
					end
					do
						local ip = _g.jk.lang.String:getSubString(domain, 1, _g.jk.lang.String:getLength(domain) - 2)
						if _g.jk.lang.String:isEmpty(ip) then
							do return false end
						end
						do
							local nodes = _g.jk.lang.String:split(ip, 46, 0)
							if not (nodes ~= nil) or _g.jk.lang.Vector:getSize(nodes) ~= 4 then
								do return false end
							end
							if nodes ~= nil then
								local n2 = 0
								local m = _g.jk.lang.Vector:getSize(nodes)
								do
									n2 = 0
									while n2 < m do
										local node = nodes[n2 + 1]
										if node ~= nil then
											ci = _g.jk.lang.String:iterate(node)
											while true do
												local c = ci:getNextChar()
												if c < 1 then
													do break end
												end
												if c >= 48 and c <= 57 then
												else
													do return false end
												end
											end
											do
												local n = _g.jk.lang.String:toInteger(node)
												if n < 0 or n > 255 then
													do return false end
												end
											end
										end
										do n2 = n2 + 1 end
									end
								end
							end
							do return false end
						end
					end
				end
				do
					local domains = _g.jk.lang.String:split(domain, 46, 0)
					if _g.jk.lang.Vector:getSize(domains) <= 1 then
						do return false end
					end
					if domains ~= nil then
						local n3 = 0
						local m2 = _g.jk.lang.Vector:getSize(domains)
						do
							n3 = 0
							while n3 < m2 do
								local label = domains[n3 + 1]
								if label ~= nil then
									if _g.jk.lang.String:isEmpty(label) then
										do return false end
									end
									if _g.jk.lang.String:getLength(label) > 63 then
										do return false end
									end
									ci = _g.jk.lang.String:iterate(label)
									if not (ci ~= nil) then
										do return false end
									end
									do
										local hyphenPreceded = false
										isFirstCharacter = true
										while true do
											local c = ci:getNextChar()
											if c < 1 then
												do break end
											end
											if c >= 65 and c <= 90 then
											elseif c >= 97 and c <= 122 then
											elseif c >= 48 and c <= 57 then
											elseif c == 45 then
												if isFirstCharacter then
													do return false end
												end
												hyphenPreceded = true
												isFirstCharacter = false
												goto _continue2
											else
												do return false end
											end
											hyphenPreceded = false
											isFirstCharacter = false
											::_continue2::
										end
										if hyphenPreceded then
											do return false end
										end
									end
								end
								do n3 = n3 + 1 end
							end
						end
					end
					do return true end
				end
			end
		end
	end
end

function jk.validator.DataValidator:checkPassword(password, minimum, maximun)
	if _g.jk.lang.String:isEmpty(password) then
		do return _g.jk.lang.Error:forCode("noPassword", nil) end
	end
	do
		local min = minimum
		if min < 8 then
			min = 8
		end
		if _g.jk.lang.String:getLength(password) < min then
			do return _g.jk.lang.Error:forCode("passwordLengthLessThanMinimum", nil) end
		end
		do
			local max = maximun
			if max < 64 then
				max = 64
			end
			if _g.jk.lang.String:getLength(password) > max then
				do return _g.jk.lang.Error:forCode("passwordLengthMoreThanMaximum", nil) end
			end
			do
				local hasLowerAlpha = false
				local hasUpperAlpha = false
				local hasNumeric = false
				local hasSpecial = false
				local previousChar = 0
				local sChar = 0
				local count = 1
				local ci = _g.jk.lang.String:iterate(password)
				while true do
					local c = ci:getNextChar()
					if c < 1 then
						do break end
					end
					if c > 32 and c < 127 then
						if previousChar == c then
							do count = count + 1 end
							if count == 3 then
								do return _g.jk.lang.Error:forCode("passwordHasRepetitiveCharacters", nil) end
							end
							previousChar = c
						elseif c == previousChar + 1 then
							sChar = c
						elseif c == sChar + 1 then
							do return _g.jk.lang.Error:forCode("passwordHasSequentialCharacters", nil) end
						else
							previousChar = c
						end
						if c >= 65 and c <= 90 then
							hasUpperAlpha = true
						elseif c >= 97 and c <= 122 then
							hasLowerAlpha = true
						elseif c >= 48 and c <= 57 then
							hasNumeric = true
						else
							hasSpecial = true
						end
						goto _continue3
					end
					do return _g.jk.lang.Error:forCode("passwordHasUnknownCharacter", nil) end
					::_continue3::
				end
				if hasUpperAlpha and hasLowerAlpha == false and hasNumeric == false and hasSpecial == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoAlphanumericCharacterCombination", nil) end
				elseif hasLowerAlpha and hasUpperAlpha and hasNumeric == false and hasSpecial == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoNumericAndSpecialCharacterCombination", nil) end
				elseif hasNumeric == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoNumericCharacter", nil) end
				elseif hasLowerAlpha == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoLowercase", nil) end
				elseif hasUpperAlpha == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoUppercase", nil) end
				elseif hasSpecial == false then
					do return _g.jk.lang.Error:forCode("passwordHasNoSpecialCharacter", nil) end
				else
					do return nil end
				end
			end
		end
	end
end

function jk.validator.DataValidator:isNameValid(name)
	if _g.jk.lang.String:isEmpty(name) then
		do return false end
	end
	do
		local ci = _g.jk.lang.String:iterate(name)
		local isFirstCharacter = true
		local spacePreceded = false
		local hyphenPreceded = false
		while true do
			local c = ci:getNextChar()
			if c < 1 then
				do break end
			end
			if c == 32 then
				if isFirstCharacter then
					do return false end
				end
				if spacePreceded then
					do return false end
				end
				spacePreceded = true
				goto _continue4
			end
			if c == 45 then
				if isFirstCharacter then
					do return false end
				end
				if hyphenPreceded then
					do return false end
				end
				hyphenPreceded = true
				goto _continue4
			end
			isFirstCharacter = false
			spacePreceded = false
			hyphenPreceded = false
			if c == 164 or c == 165 or (c >= 65 and c <= 90) or (c >= 97 and c <= 122) then
				goto _continue4
			end
			do return false end
			::_continue4::
		end
		if spacePreceded then
			do return false end
		end
		if hyphenPreceded then
			do return false end
		end
		do return true end
	end
end
